### PR TITLE
refactor NsenterWriter to utilize pkg/util/nsenter

### PR DESCRIPTION
**Pull request by dixudx**
**What this PR does / why we need it**:
Per [discussion](https://github.com/kubernetes/kubernetes/pull/51771#discussion_r138824451)
Depend on #51771

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by dixudx** [comment]
/assign @smarterclayton

PTAL

**Comment by dixudx** [comment]
ping @smarterclayton Rebased. PTAL. Thanks.

**Comment by dixudx** [comment]
ping @dims @smarterclayton PTAL. Thanks

**Comment by dims** [comment]
/approve

No idea why for just a small change, we need to get approval from so many folks ...

**Comment by dixudx** [comment]
ping @smarterclayton @brendandburns Please help review. Thanks.

**Comment by dixudx** [comment]
ping @smarterclayton @brendandburns @dchen1107  Please help review. Thanks.

**Comment by dixudx** [comment]
ping @smarterclayton @brendandburns @dchen1107 Please help review. Thanks.

**Comment by smarterclayton** [comment]
@kubernetes/sig-node-pr-reviews

**Comment by dixudx** [comment]
ping @kubernetes/sig-node-pr-reviews for review. Thanks.

**Comment by dixudx** [comment]
ping @kubernetes/sig-node-pr-reviews for review. Thanks.

**Review by feiskyer — APPROVED** [review]
/lgtm

**Comment by dixudx** [comment]
ping @smarterclayton @thockin for approval. Thanks.

File: pkg/util/io/writer.go [hunk 1]
```diff
@@ -54,25 +55,20 @@ type NsenterWriter struct{}
...
-		"--mount=/rootfs/proc/1/ns/mnt",
-		"--",
-	}
-
-	echoArgs := append(baseArgs, "sh", "-c", fmt.Sprintf("cat > %s", filename))
-	glog.V(5).Infof("Command to write data to file: %v %v", cmd, echoArgs)
-	command := exec.Command(cmd, echoArgs...)
-	command.Stdin = bytes.NewBuffer(data)
+	ne := nsenter.NewNsenter()
+	echoArgs := []string{"-c", fmt.Sprintf("cat > %s", filename)}
```
Comment (smarterclayton) [inline pkg/util/io/writer.go]: What happens if this filename has a space in it?

File: pkg/util/io/writer.go [hunk 2]
```diff
@@ -54,25 +55,20 @@ type NsenterWriter struct{}
...
 	if err != nil {
 		glog.Errorf("Output from writing to %q: %v", filename, string(outputBytes))
 		return err
 	}
 
-	chmodArgs := append(baseArgs, "chmod", fmt.Sprintf("%o", perm), filename)
-	glog.V(5).Infof("Command to change permissions to file: %v %v", cmd, chmodArgs)
-	outputBytes, err = exec.Command(cmd, chmodArgs...).CombinedOutput()
+	chmodArgs := []string{fmt.Sprintf("%o", perm), filename}
+	glog.V(5).Infof("Command to change permissions to file: nsenter --mount=/rootfs/proc/1/ns/mnt -- chmod %v", chmodArgs)
```
Comment (smarterclayton) [inline pkg/util/io/writer.go]: Hrm, I wouldn't change this.  The point of the exec is to hide that.  Would remove this specific bit.

**Comment by dixudx** [comment]
ping @feiskyer @smarterclayton Updated. PTAL. Thanks.

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Comment by dixudx** [comment]
/assign @thockin 
ping @thockin for approval. Thanks.

**Comment by dixudx** [comment]
ping @smarterclayton for approval. Thanks.

**Pull request merged**